android/java/org/chromium/chrome/browser/firstrun/WelcomeOnboardingActivity.java

```diff
@@ -408,8 +409,13 @@ private void handleWDPStep() {
         showWDPPage();
     }
 
-    private boolean isWDPEnabled() {
-        return BraveConfig.WEB_DISCOVERY_ENABLED;
+    private boolean isWDPSettingAvailable() {
+        if (!BraveConfig.WEB_DISCOVERY_ENABLED) {
+            return false;
+        }
+
+        return !UserPrefs.get(getProfileProviderSupplier().get().getOriginalProfile())
+                .isManagedPreference(WebDiscoveryPrefs.WEB_DISCOVERY_ENABLED);
```

Comment:
do you know why it's or on profile level or on local state like in the p3a and crash analytics PR? https://github.com/brave/brave-core/pull/30597/files#diff-4413cdfed46f306bb4de959a596545f6b99b1efbb7379da43e810618a19190e9R446

Reply:
that decision was made before my time here. i suspect that it was added the profile so that the "Brave extension" (which hosts the original WDP engine) can access the pref value. AFAIK, the extension `chrome.settingsPrivate` API only allows access to profile prefs, and not local state.